extensions/scarb-execute/src/lib.rs

```diff
@@ -251,6 +261,59 @@ pub fn execute(
         fs::write(air_private_input_path, output_value)?;
     }
 
+    if args.run.save_profiler_trace_data {
+        ensure!(
+            package.targets.iter().any(|t| t.kind == "executable"
```

Comment:
Checking for `any` is not enough - you need to compile **the target you are executing** to sierra. You know which target is executed from https://github.com/software-mansion/scarb/blob/c79063cbb9380a6cd4526a833dd1dc59c1638322/extensions/scarb-execute/src/lib.rs#L92